packages/core/src/validator.ts

```diff
@@ -169,16 +170,20 @@ export const createValidator = (callback: ValidationCallback, initialData: Recor
     /**
      * Create a debounced validation callback.
      */
-    const createValidator = () => debounce(() => {
-        callback({
-            get: (url, data = {}, config = {}) => client.get(url, parseData(data), resolveConfig(config, data)),
-            post: (url, data = {}, config = {}) => client.post(url, parseData(data), resolveConfig(config, data)),
-            patch: (url, data = {}, config = {}) => client.patch(url, parseData(data), resolveConfig(config, data)),
-            put: (url, data = {}, config = {}) => client.put(url, parseData(data), resolveConfig(config, data)),
-            delete: (url, data = {}, config = {}) => client.delete(url, parseData(data), resolveConfig(config, data)),
+    const createValidator = () => debounce(async (instanceConfig: Config): Promise<unknown> => callback({
+            get: (url, data = {}, globalConfig = {}) => client.get(url, parseData(data), resolveConfig(globalConfig, instanceConfig, data)),
+            post: (url, data = {}, globalConfig = {}) => client.post(url, parseData(data), resolveConfig(globalConfig, instanceConfig, data)),
+            patch: (url, data = {}, globalConfig = {}) => client.patch(url, parseData(data), resolveConfig(globalConfig, instanceConfig, data)),
+            put: (url, data = {}, globalConfig = {}) => client.put(url, parseData(data), resolveConfig(globalConfig, instanceConfig, data)),
+            delete: (url, data = {}, globalConfig = {}) => client.delete(url, parseData(data), resolveConfig(globalConfig, instanceConfig, data)),
         })
-        .catch(error => isAxiosError(error) ? null : Promise.reject(error))
-    }, debounceTimeoutDuration, { leading: true, trailing: true })
+        .catch(error => {
+            if (isAxiosError(error) && isCancel(error)) {
+                throw new RequestCancelled('An in-flight Precognition request was cancelled.', { cause: error })
+            }
```

Comment:
Could we resolve this and then never resolve the `latestPromise` instead of throwing an error?

Reply:
I'm wondering if this will fix #73 as well. This seems very much related.